server/routes/ratings.js

```diff
@@ -13,71 +14,60 @@ router.get('/', function (req, res) {
   });
 });
 
-/* GET rating by id */
-router.get('/:id', function (req, res) {
-  models.Rating.findById(req.params.id).then( function (rating) {
-    res.send(rating.toJson());
-  });
-});
-
-/* UPDATE user's rating for snippet*/
-router.put('/:id', function (req, res) {
-  models.Rating.findById(req.params.id).then( function (rating) {
-    rating.value = req.body.value;
-    rating.save({ validate: false, logging: true}).then(function () {
-      res.status(200).send('ok');
-    }).catch(function () {
-      res.status(422).send('error');
-    });
-  }).catch(function () {
-    res.status(422).send('error');
-  });
-});
-
 /* POST new rating */
 router.post('/', function (req, res) {
   var userId = req.user.get('id');
   var snippetId = req.body.SnippetId;
-  var sumRatings = 0.0;
-  models.Rating.findOne({ where : { SnippetId: snippetId, UserId: userId } }).then( function (rating) {
-    if (rating) {
-      rating.value = req.body.value;
-      rating.save({ validate: false, logging: true}).then(function () {
-        models.Rating.sum('value', { where : { SnippetId: snippetId } }).then( function (sum) {
-          sumRatings = sum;
-          models.Rating.count({ where : { SnippetId: snippetId } }).then( function (c){
-            var average = 0;
-            if(c>0){
-              average = (sumRatings/c);
-            }
-            res.status(200).send({rating: rating, avg: average.toFixed(2)});
+  var attributes = {
+    value: req.body.value,
+    UserId: userId,
+    SnippetId: snippetId
+  };
+  models.sequelize.transaction(function (t){
+    return models.Rating.findOne({ where : { SnippetId: snippetId, UserId: userId }, transaction: t }).then( function (rating) {
```

Comment:
My first thought :P
https://image.spreadshirtmedia.com/image-server/v1/designs/11785279,width=178,height=178,version=1329647643/Mother-of-God.png